apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/impl.ex

```diff
@@ -1281,20 +1281,52 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Impl do
     decoded_payload = PayloadsDecoder.safe_inflate(zlib_payload)
 
     if decoded_payload != :error do
-      operation_result = prune_device_properties(new_state, decoded_payload, timestamp_ms)
+      payload_size = byte_size(payload)
 
-      if operation_result != :ok do
-        Logger.debug("Result is #{inspect(operation_result)} further actions should be required.")
-      end
+      handle_valid_purge_properties(
+        new_state,
+        decoded_payload,
+        payload_size,
+        message_id,
+        timestamp_ms
+      )
+    else
+      Logger.warn("Invalid purge_properties payload", tag: "purge_properties_error")
+
+      {:ok, new_state} = ask_clean_session(new_state, timestamp)
+      MessageTracker.discard(new_state.message_tracker, message_id)
+
+      :telemetry.execute(
+        [:astarte, :data_updater_plant, :data_updater, :discarded_message],
+        %{},
+        %{realm: new_state.realm}
+      )
+
+      new_state
     end
+  end
 
-    MessageTracker.ack_delivery(new_state.message_tracker, message_id)
+  defp handle_valid_purge_properties(
+         state,
+         decoded_payload,
+         payload_size,
+         message_id,
+         timestamp_ms
+       ) do
+    operation_result = prune_device_properties(state, decoded_payload, timestamp_ms)
+
+    # TODO at the moment operation_result is always :ok, we might want to revise this
```

Comment:
If right now this can't fail, let's be assertive and do:
```
:ok = prune_device_properties(state, decoded_payload, timestamp_ms)
```

Otherwise we could miss if we introduce a possible error value later (a debug log will go unnoticed for a long time in production and leave a false sense of security).